enhancements/cluster-logging/container-log-throttling.md

```diff
@@ -0,0 +1,206 @@
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@redhatdan"
+creation-date: 2020-02-23
+last-updated: 2020-02-23
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - conmon PR [WIP: logging behavior policies and rated-limited logging](https://github.com/containers/conmon/pull/92)
+  - libpod PR [WIP: Added support for log policy and log rate limit in conmon](https://github.com/containers/libpod/pull/4663)
+replaces: []
+superseded-by: []
+---
+
+# Container Log Throttling
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+  Container log throttling is a set of capabilities in OpenShift aimed at providing cluster administrators with means of controlling logging policies and logging throughput of individual containers or their groupings. Containers are considered sources of two log streams, stdout and stderr, that are together routed by the container runtime to a log file on the cluster node. Stdout and stderr log streams are controlled separately.
+  The following logging policies are proposed:
+  1. Passthrough
+     Unrestricted logging. For backward compatibility, this is the default if no policy is specified. It is an error to specify a rate limit with this policy.
+  1. Ignore
+     Container stdout/stderr output is not written to the log and is discarded. It is an error to specify a rate limit with this policy.
+  1. Backpressure
+     Container is only allowed to log as fast as rate limit allows and may occasionally block to stay within that limit. A rate limit must be specified with this policy.
+  1. Drop
+     Container is only allowed to log as fast as rate limit allows. The container avoids blocking, but some log output may be skipped and not written to the log to stay within the rate limit. A rate limit must be specified with this policy.
+
+   The log stream rate limit is specified in bytes per second. A suffix of "K", "M", "G", or "T" can be added to denote kibibytes (*1024), mebibytes, and so on.
+
+## Motivation
+
+  Administrators of an OpenShift cluster need the proper controls/policies for logging behaviors in order to maintain a balanced system. Without these controls, the rate at which containers can emit logs can exceed the rate at which logging collectors can read them, leading to a situation where the kubelet deletes entire log files as part of log rotation before they are read by the collectors. This leads to multiple megabytes of logs being lost.
+
+## Open Questions
+
+1. How can we help cluster administrators identify containers that require log throttling?
+1. How can groupings of containers can be expressed in OpenShift for the purposes of log throttling?
```

Comment:
Wouldnt we take advantage of  label mechanisms for to group containers

Reply:
AFAIK we can't label containers as they are not k8s API objects.